x-pack/plugins/security/index.js

```diff
@@ -65,15 +65,10 @@ export const security = (kibana) => new kibana.Plugin({
       audit: Joi.object({
         enabled: Joi.boolean().default(false)
       }).default(),
-      authc: Joi.object({})
-        .when('authProviders', {
-          is: Joi.array().items(Joi.string().valid('oidc').required(), Joi.string()),
-          then: Joi.object({
-            oidc: Joi.object({
-              realm: Joi.string().required(),
-            }).default()
-          }).default()
-        })
+      authc: Joi.object({
```

Comment:
__note:__ couldn't find a good way to reproduce the Joi-based check we have in master for `oidc` so that it covers both `saml` and `oidc`. So had to spread this check over multiple places and not support happy about that :/ If you have any suggestions, I'm all ears!

Reply:
Once again `Joi` is a real pain, I got close to doing this using `Joi`, but it doesn't work when using both `oidc` and `saml`. I'm not opposed to the way you have it right now, I wish it was earlier in the Kibana server lifecycle, but it's tolerable the way it is now.

Reply:
> I wish it was earlier in the Kibana server lifecycle

Yeah, in NP, we'll have this check in a custom validator function attached to the main config schema. 

__But__, maybe it's time to move `authProviders` to `authc.providers` (I planned this since oidc PR added `authc` node)? If you're fine with that, we can go with something like this instead (not sure if it plays well with kbn/config-schema, but I can improve that if needed):
```ts
...
config(Joi) {
  const providerOptionsSchema = (providerName, schema) => Joi.any()
    .when('providers', {
      is: Joi.array().items(Joi.string().valid(providerName).required(), Joi.string()),
      then: schema,
      otherwise: Joi.any().forbidden(),
    });

    return Joi.object({
      ....
      authc: Joi.object({
        providers: Joi.array().items(Joi.string()).default(['basic']),
        oidc: providerOptionsSchema('oidc', Joi.object({ realm: Joi.string().required() }).required()),
        saml: providerOptionsSchema('saml', Joi.object({ realm: Joi.string() })),
      })
    }).default();
  },
},

deprecations: function ({ unused, rename }) {
  return [
    .....
    rename('authProviders', 'authc.providers'),
  ];
}
...
```

Reply:
> But, maybe it's time to move authProviders to authc.providers (I planned this since oidc PR added authc node)? If you're fine with that, we can go with something like this instead (not sure if it plays well with kbn/config-schema, but I can improve that if needed):

I like this idea the best!

Reply:
Awesome, I'll make these changes and ask for review once everything is ready :)